src/main/java/uk/gov/ons/ctp/integration/rhsvc/service/impl/CaseServiceImpl.java

```diff
@@ -258,4 +301,37 @@ private void validateContactName(Contact contact) throws CTPException {
               + "'forename' and 'surname'");
     }
   }
+
+  private Optional<CaseDTO> getLatestCaseByUPRN(UniquePropertyReferenceNumber uprn)
+      throws CTPException {
+    Optional<CaseDTO> result = Optional.empty();
+
+    Optional<CollectionCase> caseFound =
+        dataRepo.readLatestCollectionCaseByUprn(Long.toString(uprn.getValue()));
+    if (caseFound.isPresent()) {
+      log.with("case", caseFound.get().getId())
+          .with("uprn", uprn)
+          .debug("Existing case found by UPRN");
+      result = Optional.of(mapperFacade.map(caseFound.get(), CaseDTO.class));
+    }
+
+    return result;
+  }
+
+  private CaseType determineCaseType(CaseRequestDTO request) {
```

Comment:
This looks identical to **determinePrimaryCaseType** in **UniqueAccessCodeServiceImpl** . Worth putting a method in **ServiceUtil** for reuse.

Reply:
This code is indeed very similar to the determineAddressLevel() in UniqueAccessCodeServiceImpl, and if was then the same then there should be a common method, but the differences mean that I don't think it makes sense to merge them into one method. It would have to take a UAC (or null) and some sort of mode enum to control which if statement gets used. The medicine would be worse than the illness.

Reply:
Hi Pete, I think the two methods I am comparing are identical apart from the method name and which class they live in. Maybe you are thinking of something different, since you reference determineAddressLevel() above, which is not the one I am talking about.

Reply:
Please disregard my earlier comment. I completely misread Rob's point.
Common code now extracted to ServiceUtil.
Good spot.